lib/arch/neon64/codec.c

```diff
@@ -41,6 +39,43 @@ static const char *base64_table_enc_transposed =
 	"Oeu+"
 	"Pfv/"
 };
+
+// Decoding
+// The input consists of five valid character sets in the Base64 alphabet,
+// which we need to map back to the 6-bit values they represent.
+// There are three ranges, two singles, and then there's the rest.
+//
+//   #  From       To        LUT  Characters
+//   1  [0..42]    [255]      #1  invalid input
+//   2  [43]       [62]       #1  +
+//   3  [44..46]   [255]      #1  invalid input
+//   4  [47]       [63]       #1  /
+//   5  [48..57]   [52..61]   #1  0..9
+//   6  [58..63]   [255]      #1  invalid input
+//   7  [64]       [255]      #2  invalid input
+//   8  [65..90]   [0..25]    #2  A..Z
+//   9  [91..96]   [255]      #2 invalid input
+//  10  [97..122]  [26..51]   #2  a..z
+// (11) Everything else => invalid input
+
+// First LUT will use VTBL (out of range indices are set to 0 in destination)
+static const uint8_t base64_dec_lut1[] =
+{
+	255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U,
+	255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U,
+	255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U, 255U,  62U, 255U, 255U, 255U,  63U,
+	 52U,  53U,  54U,  55U,  56U,  57U,  58U,  59U,  60U,  61U, 255U, 255U, 255U, 255U, 255U, 255U
+};
+// Second LUT will use VTBX (out of rang indices will be unchanged in destination)
```

Comment:
Thanks for the explanation, it would be good to add some more comments. I was just in the process of answering the question "but what happens when a character is > 126?", and what you wrote would have helped me find the answer. (My NEON skills are a bit rusty.)